src/UriNormalizer.php

```diff
@@ -207,11 +209,13 @@ private static function decodeUnreservedCharacters(UriInterface $uri): UriInterf
             return rawurldecode($match[0]);
         };
 
-        /** @var string $pathStr */
         $pathStr = preg_replace_callback($regex, $callback, $uri->getPath());
-        /** @var string $queryStr */
         $queryStr = preg_replace_callback($regex, $callback, $uri->getQuery());
 
+        /**
+         * @var string $pathStr
+         * @var string $queryStr
```

Comment:
What you had before was fine in this instance.

Reply:
Ok, let me try my best.